Prepare the changelog for v0.3.0. (#273)
hasura/ndc-postgres: changelog.md

## [Unreleased]

## [0.3.0] - 2023-01-31

### Added

- Basic support for auto-generated insert procedures for tables.
  ([#261](https://github.com/hasura/ndc-postgres/pull/261))

> - Support for composite types in inputs and outputs. Transformations are not yet
>   supported.

Reviewer: 'Transformations'? Perhaps you meant:

```suggestion
- Support for composite types in inputs and outputs. Automatic introspection
  is not yet supported.
```

Author: I meant that we don't provide any operators on these composite types. I will add both.

```suggestion
- Support for composite types in inputs and outputs. Right now, no operators are
  provided for composite types, and there is no automatic introspection.
```

Reviewer: Hm.. I still wonder if it's relevant to say that, since the ndc spec does not allow for any operators on things that are not scalar types.

Wording it like this suggests this is just something that has not been implemented in ndc-postgres yet.

Author: I am personally OK with that. I think it's more important to state that we do or don't do something than _why_.

If someone cares, they will ask!

  ([#240](https://github.com/hasura/ndc-postgres/pull/240))
- Support for composite types and arrays in variables.
  ([#249](https://github.com/hasura/ndc-postgres/pull/249))
  ([#263](https://github.com/hasura/ndc-postgres/pull/263))
- When explaining a query without variables (such as when joining across a
  remote relationship), explaining now still provides the generated SQL, and
  omits the `EXPLAIN` result from PostgreSQL.
  ([#241](https://github.com/hasura/ndc-postgres/pull/241))

### Changed

- Native query mutations are now marked as procedures instead of collections
  ([#234](https://github.com/hasura/ndc-postgres/pull/234))
- Introspection now avoids `DISTINCT ON` to be more compatible with PostgreSQL variants
  ([#260](https://github.com/hasura/ndc-postgres/pull/260))

## [0.2.0] - 2023-12-21

> ### Added

Author: We have two "Added" sections for v0.2.0. Not sure why. I will open a separate PR to change this.

Reviewer 2: The item listed under the top "Added" section of v0.2.0 are changed that are supposed to be in the changelog for v0.3.0

Reviewer 2: I probably added it in [this pr](https://github.com/hasura/ndc-postgres/commit/dd4172889830ecb5c6362e0be90b9c1106574363#diff-3bd14d078188074c410028847113ceae68865d0ad5b844a27183ef87fbe2fcc3) by mistake instead of above the v0.2.0 header.

Author: Oh, you're right! I will update that. I see I actually duplicated some of them too.
